### PR TITLE
Fix memory leak

### LINKED ISSUE #949: GUI v2 seems to be leaking memory
Initially venus-gui-v2 takes ~80MB of unique memory, but can grow to consume most of the memory (e.g. 953m or 95%) before forced reboot.

### PR DESCRIPTION
**Pull request by DanielMcInnes**

### CODE REVIEW

**Review by chriadam — DISMISSED** [review]
LGTM, although I doubt the engine calls instance() more than once to instantiate the singleton.  See e.g. https://doc.qt.io/qt-6/qml-singleton.html#defining-singletons-in-c example where the instance() method returns a new instance.

**Comment by chriadam** [comment]
Yikes!  Good catch, thanks.

File: src/units.cpp [hunk 1]
```diff
@@ -40,7 +40,8 @@ namespace Units {
 
 QObject* Units::instance(QQmlEngine *, QJSEngine *)
 {
-	return new Units;
+	static QObject* units = new Units();
```
Comment (chriadam) [inline src/units.cpp:43]: it looks line we don't actually need the engine ptr, so the Units type should be default constructible.  See https://doc.qt.io/qt-6/qml-singleton.html#defining-singletons-in-c - I suggest we get rid of the instance method, if possible.

Comment (chriadam) [reply src/units.cpp:43]: Ah, it's for QuantityInfo which calls instance() manually.

Comment (chriadam) [reply src/units.cpp:43]: style nitpick: prefer just `static QObject *units = new Units;  return units;`

**Review by chriadam — APPROVED** [review]
One style nitpick but LGTM.

File: src/units.cpp (hunk 1 above)
Comment (DanielMcInnes) [reply src/units.cpp]: Done. OOI, Why do we prefer:
`static QObject *units = new Units; `
over:
`static QObject* units = new Units();`
? Google won't tell me

**Pull request merged**